qiskit/circuit/quantumcircuit.py

```diff
@@ -1843,31 +1850,43 @@ def parameters(self):
 
         # parameters in global phase
         if isinstance(self.global_phase, ParameterExpression):
-            return params.union(self.global_phase.parameters)
+            params = params.union(self.global_phase.parameters)
+
+        # sort the parameters
+        sorted_by_name = sorted(list(params), key=functools.cmp_to_key(_compare_parameters))
 
-        return params
+        # return as parameter view, which implements the set and list interface
+        return ParameterView(sorted_by_name)
 
     @property
     def num_parameters(self):
         """Convenience function to get the number of parameter objects in the circuit."""
         return len(self.parameters)
 
-    def assign_parameters(self, param_dict, inplace=False):
+    @deprecate_arguments({'param_dict': 'params'})
+    def assign_parameters(self, params, inplace=False, *,
```

Comment:
Do we want to call this `parameters` instead of `params`, actually? It would fit better to the method name (assign _parameters_) and to the method that returns the parameters: `QuantumCircuit.parameters`.

Reply:
Yes